migration/src/m0001140_expand_spdx_licenses_function/expand_up.sql

```diff
@@ -0,0 +1,22 @@
+CREATE OR REPLACE FUNCTION expand_license_expression(
+    license_text TEXT,
+    sbom_id_param UUID
+) RETURNS TEXT AS $$
+DECLARE
+    result_text TEXT := license_text;
+    license_mapping RECORD;
+BEGIN
+    -- Replace each license reference with its corresponding name
+    FOR license_mapping IN
+        SELECT license_id, name
+        FROM licensing_infos
+        WHERE sbom_id = sbom_id_param
+    LOOP
+        -- Replace the license_id with the license name in the expression
+        -- This handles exact matches and license references within expressions
+        result_text := regexp_replace(result_text, '\m' || license_mapping.license_id || '\M', license_mapping.name, 'g');
+END LOOP;
+
+RETURN result_text;
+END;
+$$ LANGUAGE plpgsql;
```

Comment:
IIRC, there had been several keywords like `parallel` which had a performance impact.

@JimFuller-RedHat can you help out here?

Reply:
I've added `STABLE PARALLEL SAFE` following what we've been doing for other functions:
- `STABLE` because it will return the same result within a single statement execution, but the result could change between different executions if the `licensing_infos` table data is modified
- `PARALLEL SAFE` because the function only reads from `licensing_infos` table, processes text locally and returns a result without any database modifications. All operations are safe for concurrent execution.

Reply:
@JimFuller-RedHat I'm moving forward with merging this PR because it blocks some other tasks but please, feel free to come back to me with any suggestion/comment you find and I'll enhance this in a separated PR.